server.go

```diff
@@ -33,16 +43,20 @@ func NewServer(opts ServerOpts) (*http.Server, error) {
 		return nil, err
 	}
 
-	// Use logging
-	handler = handlers.LoggingHandler(os.Stderr, handler)
-
 	// Use authentication if username provided
 	if len(opts.Auth.Name) > 0 {
 		handler = web.BasicAuthMiddleware(opts.Auth, handler)
 	}
 
+	// Attach to main router
+	r.PathPrefix("/").Handler(handler)
+
+	// Use logging
+	handler = handlers.LoggingHandler(os.Stderr, r)
+
+
 	return &http.Server{
 		Addr:    opts.Port,
-		Handler: handler,
+		Handler: r,
```

Comment:
It should be `handler` here not `r`, since `handler` is derived from `r` when applying the `LoggingHandler`